dashboard/app/controllers/api_controller.rb

```diff
@@ -11,6 +11,87 @@ def user_hero
     head :not_found unless current_user
   end
 
+  def update_lockable_state
+    updates = params.require(:updates)
+    updates.to_a.each do |item|
+      user_level_data = item[:user_level_data]
+
+      if user_level_data[:user_id].nil? || user_level_data[:level_id].nil? || user_level_data[:script_id].nil?
+        raise 'Must provide user, level, and script ids'
+      end
+
+      if item[:locked] && item[:view_answers]
+        raise 'Can not view answers while locked'
+      end
+
+      unless User.find(user_level_data[:user_id]).teachers.include? current_user
+        raise 'Can only update lockable state for users students'
+      end
+
```

Comment:
Above here is validation that makes sense in the controller.  Can the logic below here be moved to a method on the `UserLevel` model (and tested in a rake test)?